snuba/utils/streams/backends/local/backend.py

```diff
@@ -92,8 +86,10 @@ def subscribe(
 
             assignment: MutableMapping[Partition, int] = {}
 
-            for topic in self.__topics.keys() & set(topics):
```

Comment:
Kafka will actually throw if you try to subscribe to a topic that doesn't exist:

```
KafkaError{code=UNKNOWN_TOPIC_OR_PART,val=3,str="Subscribed topic not available: invalid: Broker: Unknown topic or partition"}
```

This brings it more in line with the Kafka behavior, albeit with a different exception type. These may be made consistent in the future (mostly for the sake of cross-testing to avoid inconsistencies in the future) but I don't think anybody is catching this and trying to recover from it, so as long as they both throw that's good enough for me right now.